### ISSUE TITLE
[epic] Integrate indices within Kazana

### ISSUE THREAD

**Issue by patriciagarcia** [eHealthAfrica/kazana#30]
Find a way to use indices thresholds and indicator targets within the CouchDB design documents without hard coding them.

It should be possible to edit targets and thresholds through an API.

**Comment by jo** [comment]
Maybe I can I help here?

**Comment by gr2m** [comment]
The basic idea is, that indices are design docs with map/reduce functions. Targets / thresholds are meta properties on the design docs that can be edited by app users.

That creates a challenge for bootstrapping these design docs, as we don't want the user-set targets / thresholds to be overwritten when the app starts

**Comment by jo** [comment]
Could the users make copies of those ddocs?

**Comment by jo** [comment]
Surely the upgrade then must be handled if the source ddoc changes. But an upgrade must be handled anyway.

**Comment by gr2m** [comment]
Yeah I wonder if we could defined one property like `userData` which does not get overwritten during bootstrap

**Comment by jo** [comment]
And the targets / thresholds cannot be included in the query?

**Comment by gr2m** [comment]
good question :) Targets / thresholds would be relevant for reduce functions only anway, so yeah, why not put them into query, good idea :bulb: :ok_hand:

**Comment by ehealth-waffle** [comment]
I think the right way to go with "configurable views" is to separate these two things. Use entities for configuration, have the view be only the logic, and have kazana combine the two on request. That way we work around the bootstrapping issue

**Comment by gr2m** [comment]
I think the right way to go with "configurable views" is to separate these two things. Use entities for configuration, have the view be only the logic, and have kazana combine the two on request. That way we work around the bootstrapping issue
